reliability/single-fit/trace/src/test/kotlin/com/datadog/android/trace/integration/otel/OtelTracerProviderTest.kt

```diff
@@ -0,0 +1,984 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.trace.integration.otel
+
+import com.datadog.android.api.feature.Feature
+import com.datadog.android.core.stub.StubSDKCore
+import com.datadog.android.tests.ktx.getInt
+import com.datadog.android.trace.OtelTracerProvider
+import com.datadog.android.trace.Trace
+import com.datadog.android.trace.TraceConfiguration
+import com.datadog.android.trace.integration.opentracing.AndroidTracerTest
+import com.datadog.android.trace.integration.tests.assertj.SpansPayloadAssert
+import com.datadog.android.trace.integration.tests.elmyr.TraceIntegrationForgeConfigurator
+import com.datadog.opentelemetry.trace.OtelConventions
+import com.datadog.tools.unit.extensions.TestConfigurationExtension
+import com.datadog.trace.api.sampling.PrioritySampling
+import com.datadog.trace.bootstrap.instrumentation.api.Tags
+import com.google.gson.JsonObject
+import com.google.gson.JsonParser
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.annotation.DoubleForgery
+import fr.xgouchet.elmyr.annotation.StringForgery
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import io.opentelemetry.api.common.AttributeKey
+import io.opentelemetry.api.common.Attributes
+import io.opentelemetry.api.trace.Span
+import io.opentelemetry.api.trace.SpanKind
+import io.opentelemetry.context.Context
+import io.opentelemetry.context.ContextStorage
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.data.Offset
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.RepeatedTest
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.quality.Strictness
+import java.time.Instant
+import java.util.concurrent.TimeUnit
+import kotlin.system.measureNanoTime
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class),
+    ExtendWith(TestConfigurationExtension::class)
+)
+@ForgeConfiguration(TraceIntegrationForgeConfigurator::class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+internal class OtelTracerProviderTest {
+
+    private lateinit var stubSdkCore: StubSDKCore
+
+    @BeforeEach
+    fun `set up`(forge: Forge) {
+        val datadogContextStorageClass =
+            Class.forName("com.datadog.android.trace.opentelemetry.DatadogContextStorage")
+        val constructor = datadogContextStorageClass.constructors.first { it.parameterCount == 1 }
```

Comment:
`constructor` is a reserved keyword in Kotlin, even github highlights this word differently. To avoid confusion it is better to rename it to something like `ctor` or `contructorRef`.